backend/corpora/lambdas/api/v1/curation/collections/common.py

```diff
@@ -94,8 +95,16 @@ def reshape_dataset_for_curation_api(dataset: dict, preview=False) -> dict:
         for asset in artifacts:
             if asset["filetype"] in (DatasetArtifactFileType.H5AD, DatasetArtifactFileType.RDS):
                 dataset["dataset_assets"].append(asset)
-    if dataset.get("processing_status"):
-        dataset["processing_status"] = dataset["processing_status"]["processing_status"]
+    if processing_status := dataset.pop("processing_status", None):
+        if processing_status["processing_status"] == ProcessingStatus.FAILURE:
+            if processing_status["validation_status"] == ValidationStatus.INVALID:
+                dataset["processing_status_detail"] = processing_status["validation_message"]
+                dataset["processing_status"] = "VALIDATION_FAILURE"
+            else:
+                dataset["processing_status"] = "PIPELINE_FAILURE"
+                dataset["processing_status_detail"] = "Someone from the CELLxGENE support team will reach out to you."
```

Comment:
we should update the open API spec to include processing_status_detail as part of the dataset response schema